apps/wallet/src/ui/app/shared/Portal.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+type PortalProps = {
+    children: React.ReactNode;
+    containerId: string;
+};
+
+export function Portal({ children, containerId }: PortalProps) {
+    const [hasMounted, setHasMounted] = useState(false);
```

Comment:
do we need this?

Reply:
there are several stacking contexts in the main page layout which makes things tricky to deal with when you consider modals, overlays, toasts, menus that appear over specific content, navigation tabs, etc. I originally tried to come up with a workable solution 1). not using portals and 2). portaling content to DOM elements that exist in the `index.html` but that came with some of its own issues. 

I'll admit I more-or-less copy + pasta'd this from https://www.joshwcomeau.com/snippets/react-components/in-portal/  haha, but you can't render the Portal until after the initial render since the portal container element won't exist in the DOM during the first render

Reply:
ah yea, I normally just throw the portal container in `index.html` which has been pretty painless for me in the past

Reply:
```
<body> // width: 100%, height: 100%, full-screen BG image
  <div id="app-wrapper"> // width: 360px, height: 595px
    <div id="root">React tree rendered here</div>
    <div id="overlay-portal-container"></div> // absolutely positioned
    <div id="toaster-portal-container"></div> // absolutely positioned
  </div>
</body>
```

I thought of doing something like the above, but then I was kinda iffy since I didn't want to write too much non-React HTML/CSS to layout the page correctly 🤷🏼. I'm def open to ideas if you have a better one haha. This was the other idea, but it kind of sucks to have to hardcode the width and height in several different places

```
<body> // width: 100%, height: 100%, full-screen BG image
    <div id="root">React tree rendered here</div>
    <div id="overlay-portal-container"></div> // absolutely positioned, width: 360px, height: 595px
    <div id="toaster-portal-container"></div> // absolutely positioned, width: 360px, height: 595px
</body>
```

Reply:
yea, makes sense. the constrained size of the wallet layout makes things a little tricker. it would be nice to reign in the z-indices at some point, I don't see _too_ many declarations in wallet